src/libvirtApi/network.js

```diff
@@ -33,6 +33,21 @@ export function networkActivate({ connectionName, objPath }) {
     return call(connectionName, objPath, 'org.libvirt.Network', 'Create', [], { timeout, type: '' });
 }
 
+export function networkAddStaticHostEntries({ connectionName, networkName, macAddress, ipAddress, parentIndex }) {
+    const opts = { err: "message", environ: ['LC_ALL=C.UTF-8'] };
+    if (connectionName === 'system')
+        opts.superuser = 'try';
+
+    const cmd = [
+        'virsh', '-c', `qemu:///${connectionName}`,
+        'net-update', networkName,
+        'add-last', 'ip-dhcp-host', `<host mac='${macAddress}' ip='${ipAddress}' />`,
+        '--parent-index', parentIndex,
+        '--config'
+    ];
+    return cockpit.spawn(cmd, opts);
```

Comment:
Why not instead use the proper libvirt-dbus API which has implementation for: https://libvirt.org/html/libvirt-libvirt-network.html#virNetworkUpdate
I think they are equivalent with the virsh net-update.

Reply:
Goood point :) did not see it!